### PR TITLE
Kodi 19/Python 3 compatibility

### PR DESCRIPTION
**Pull request by pannal**
GHI (If applicable): #

## Description:
Kodi 19 drops Python 2 in favor of Python 3. This uses kodi.six and the six module, as well as a couple of compatibility changes in the code, to be backwards compatible with Kodi 18 and Python 3 compatible.

## Checklist:
- [x] I have based this PR against the develop branch

### CODE REVIEW

**Comment by ruuk** [comment]
You can probably drop that last commit and rebase (since I committed a 'proper' fix). I'll try to review and test this soon.

**Comment by pannal** [comment]
Ah, this isn't backwards compatible with Kodi 18 anymore btw. They fully dropped Py2 support.

File: addon.xml [hunk 1]
```diff
@@ -4,8 +4,10 @@
        version="0.1.7a3"
        provider-name="Plex">
     <requires>
-        <import addon="xbmc.python" version="2.25.0"/>
-        <import addon="script.module.requests" version="2.3.0"/>
+        <import addon="xbmc.python" version="3.0.0"/>
```
Comment (ruuk) [inline addon.xml]: This should remain at the old version. I can push to the Kodi 19 repo with just this version change.

**Comment by ruuk** [comment]
The python 3 compatibility PR #307 (Kodi 18) has been merged. I'll maintain the [develop-kodi19](https://github.com/plexinc/plex-for-kodi/tree/develop-kodi19) branch which is simply the develop branch with a single commit for the `addon.xml` version changes that I can just keep rebased and use to submit for Kodi 19 in addition to the normal submission.

Thanks for the compatibility work! I really appreciate it.

**Pull request closed without merging**